extensions/community_extensions/dsa_transparency.md

```diff
@@ -0,0 +1,345 @@
+<td>Extension for DSA transparency information</td>
+</tr>
+</tbody>
+</table>
+</div>
+<h4>Object: DSA</h4>
+<div>
+<table>
+<tbody>
+<tr>
+<td><strong>Attribute</strong></td>
+<td><strong>Type</strong></td>
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td><code>dsainfo</code></td>
+<td>integer</td>
+<td>Flag to indicate if DSA information should be made available. This will signal if the bid request belongs to an Online Platform/VLOP, such that a buyer should respond with DSA Transparency information based on the pubrender value.
+    <p><code>0</code> = Not required</p>
+    <p><code>1</code> = Supported, bid responses with or without DSA object will be accepted</p>
+    <p><code>2</code> = Required, bid responses without DSA object will not be accepted</p>
+    <p><code>3</code> = Required, bid responses without DSA object will not be accepted, Publisher is an Online Platform</p></td>
+</tr>
+<tr>
+<td><code>pubrender</code></td>
+<td>integer</td>
+<td>Flag to indicate if the publisher will render the DSA Transparency info. This will signal if the publisher is able to and intends to render the icon and display the DSA transparency info to the end user.
+    <p><code>0</code> = Publisher can't render</p>
+    <p><code>1</code> = Publisher could render depending on adrender</p>
+    <p><code>2</code> = Publisher will render</p></td>
+</tr>
+<tr>
+<td><code>datatopub</code></td>
+<td>integer</td>
+<td>Independent of pubrender, the publisher may need the transparency data for audit purposes.
+    <p><code>0</code> = do not send transparency data</p>
+    <p><code>1</code> = optional to send transparency data</p>
+    <p><code>2</code> = send transparency data</p></td>
+</tr>
+<tr>
+<td><code>transparency</code></td>
+<td>array of object</td>
+<td>Array of objects of the entities that applied user parameters and the parameters they applied.</td>
+</tr>
+</tbody>
+</table>
+</div>
+<h4>Object: Transparency&nbsp;</h4>
+<div>
+<table>
+<tbody>
+<tr>
+<td><strong>Attribute</strong></td>
+<td><strong>Type</strong></td>
+<td><strong>Description</strong></td>
+</tr>
+<tr>
+<td><code>domain</code></td>
+<td>string&nbsp;</td>
+<td>Domain of the entity that applied user parameters&nbsp;</td>
+</tr>
+<tr>
+<td><code>params</code></td>
+<td>array of integer</td>
+<td>Array for platform or sell-side use of any user parameters (using <a href="#user_parameters" target="_blank" rel="noopener">the list provided by DSA Transparency Taskforce</a>). <em>Note; See definition and list of possible </em><a href="#user_parameters" target="_blank" rel="noopener"><em>user parameters as listed here</em></a><em>, applied consistently in both bid request and/or bid response.</em></td>
+</tr>
+</tbody>
+</table>
+</div>
+<h4>Sample OpenRTB 2.6 Bid Request with DSA transparency:</h4>
+
+```
+{
+    "id": "80ce30c53c16e6ede735f123ef6e32361bfc7b22",
+    "at": 1,
+    "cur": [
+   	 "USD"
+    ],
+    "regs": {
+   	 "ext": [{
+   		 "dsa": [{
+            "dsainfo": 3, 
```

Comment:
Do we need to repeat the `dsa` name in `dsainfo`? Can it simply be `info`?

Reply:
The GPWG reviewed and has since updated this to `required` since info could be misleading as this indicates whether the dsa information is needed and not the actual dsa information itself.

Reply:
`required` is considered as a unfortunate naming as well, since keyword `required` is a reserved one in many programming languages as well as in protobuf v2 and thrift:

proto v2 https://protobuf.dev/reference/protobuf/proto2-spec/#fields
thrift https://thrift.apache.org/docs/idl.html#field-requiredness
Please consider to rename it to something else, otherwise it will be very painful to implement json parsing.